docroot/modules/custom/nvli_custom/nvli_custom.module

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * @file
+ * Contains nvli_custom.module.
+ */
+
+/**
+ * Implements hook_mirador_metadata_alter.
+ */
+function nvli_custom_mirador_metadata_alter(&$entity) {
+  // field_solr_doc_id contains the solr doc id, corresponding
+  // to this entity.
+  $solr_doc_id = $entity->field_solr_doc_id->value;
+  // @to-do: Get the solr server name from variable.
+  $solr_server = "nvli";
+  $metadata = \Drupal::service('custom_solr_search.search')->basicSearch($solr_doc_id, 0, 5, $solr_server);
```

Comment:
Modified the method. Check this https://github.com/NVLI/nvli-main-website/pull/13